t/plugin/opa2.t

```diff
@@ -0,0 +1,217 @@
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/u1',
+                ngx.HTTP_PUT,
+                [[{
+                    "nodes": {
+                        "127.0.0.1:1980": 1
+                    },
+                    "type": "roundrobin"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 2: setup consumer
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/consumers',
+                ngx.HTTP_PUT,
+                [[{
+                    "username": "test",
+                    "plugins": {
+                        "key-auth": {
+                            "disable": false,
+                            "key": "test-key"
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: setup service
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/services/s1',
+                ngx.HTTP_PUT,
+                [[{
+                    "name": "s1",
+                    "plugins": {
+                        "key-auth": {
+                            "disable": false
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 4: setup route with APISIX data
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "opa": {
+                            "host": "http://127.0.0.1:8181",
+                            "policy": "example2",
+                            "with_route": true,
+                            "with_upstream": true,
+                            "with_consumer": true,
+                            "with_service": true
+                        }
+                    },
+                    "upstream_id": "u1",
+                    "service_id": "s1",
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 5: hit route (test without apikey)
```

Comment:
This one seems is not relative to OPA?

Reply:
It is used to check if the service and consumer are in effect on the route. I have removed it.